### ISSUE TITLE
Ensure that different data pipelines are logically separate

### ISSUE THREAD

**Issue by jkarpen** [cagov/caldata-mdsa-caltrans-pems#375]

**Comment by pingpingxiu-DOT-ca-gov** [comment]
The problem Ian stated was valid, in that data relay was not designed to be "multi-tasking," more specifically, it can only sequentially execute the planned tasks. Therefore, if the plan changes, such as in the backfill situation where we need to go back in history to redo the relay, the current data relay has to choose either completely cancel current existing plans and focus on the backfill, or continue to execute the existing plan all the way until its completion, after which other tasks are possible to execute.

The root cause is the single queue design. 

![Single Queue Data Relay drawio](https://github.com/user-attachments/assets/471ed981-88b3-4c1e-830e-37c03c6ce272)

To mitigate the situation that the last task wait too long, I change the design to be multi-task-queue like below: 

![Multi Queue Data Relay drawio](https://github.com/user-attachments/assets/0eb066cd-9467-471f-b23e-5d6e0e77905e)

**Comment by pingpingxiu-DOT-ca-gov** [comment]
Thanks for the thoughtful discussion. 
Document the QnA: 
1. Is this the only solution for the challenge? 
Answer: the other solution would be to increase the machines (or processes) that pulls the data base, therefore the task queue has a faster speed to complete. The problem is, still single thread, and also, increasing the data pull will exert pressure on the production database. 

2. Would this address the "logic separation" concern? 
Answer: Yes. The tasks from 30sec relay, daily config relay and the backfill will have their fair treatment, due to the fact that they are no longer assigned to a single queue, and their respective queue will have guaranteed handling from the data puller.

**Comment by jkarpen** [comment]
Pingping successfully implemented the changes outlined above. As a next step before closing this issue Pingping will like to implement a dashboard to track performance of the different queues. Pingping will meet with @ian-r-rose when he returns next week for input on the KPIs to use.

**Comment by jkarpen** [comment]
Next step on this task is to document the code. Pingping will create a separate issue for creating a dashboard to track performance.

**Comment by jkarpen** [comment]
Per @pingpingxiu-DOT-ca-gov there is not a need to add a new dashboard, existing dashboards should capture any issues. Next step on this issue is for Pingping and @ian-r-rose to meet and review the code.

**Comment by jkarpen** [comment]
Per @pingpingxiu-DOT-ca-gov this is waiting on the virtual environments PR to be completed. Then Pingping will submit a PR for this to be further reviewed before completion.
